mlrun/runtimes/base.py

```diff
@@ -1121,10 +1170,84 @@ def _is_runtime_resource_run_in_transient_state(
 
         return False, last_update
 
+    def _verify_run_reached_stable_state_and_monitor_progress(
+        self, db: DBInterface, db_session: Session, project: str, run_uid: str
+    ):
+        """
+        This function runs in a retry_until_successful and therefore need to raise if run didn't reach stable state
+        """
+        desired_run_state = self._resolve_run_state_representing_current_resource_state(
+            db, db_session, project, run_uid
+        )
+        updated_run_state = self._ensure_run_status_updated(
+            db, db_session, project, run_uid, desired_run_state
+        )
+
+        if updated_run_state not in RunStates.stable_states():
+            raise Exception("run did not reach stable state yet")
+
+    def _resolve_run_state_representing_current_resource_state(
```

Comment:
```suggestion
    def _resolve_run_state(
```
🤷  am i missing something?

Reply:
I would expect function like `_resolve_run_state` to go to the DB, query for the run, find the state and return it.
While this function is actually checking the runtime resource, resolving its state, and translating it to what should be the representing run state

Reply:
This function does not exist anymore (as part of moving to one loop to monitor all runs)

Reply:
`_resolve_current_run_state` would've addressed you first comment (what you describe sounds more like a "get" or "fetch" to me)
But, like you wrote, not relevant now I suppose